dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -73,7 +73,8 @@
 
 // Ensure that the version requested belongs to a workflow a user has access to.
 @NamedQueries({
-        @NamedQuery(name = "io.dockstore.webservice.core.Version.findVersionInEntry", query = "SELECT v FROM Version v WHERE :entryId = v.parent.id AND :versionId = v.id")
+        @NamedQuery(name = "io.dockstore.webservice.core.Version.findVersionInEntry", query = "SELECT v FROM Version v WHERE :entryId = v.parent.id AND :versionId = v.id"),
+        @NamedQuery(name = "io.dockstore.webservice.core.Version.getCountByEntryId", query = "SELECT Count(v) FROM Version v WHERE v.parent.id = :id")
```

Comment:
i'll double check

Reply:
the resulting sql is identical:

`org.hibernate.SQL: /* io.dockstore.webservice.core.Version.getCountByEntryId */ select count(version0_.id) as col_0_0_ from ( select id, commitID, dbCreateDate, dbUpdateDate, dirtyBit, frozen, name, reference, referenceType, valid, parentid, versionEditor_id, automated, cwlPath, dockerfilePath, imageId, lastBuilt, size, wdlPath, null::varchar as dagJson, null::boolean as isLegacyVersion, null::timestamp as lastModified, null::int4 as subClass, null::varchar as toolTableJson, null::varchar as workflowPath, 1 as clazz_ from tag union all select id, commitID, dbCreateDate, dbUpdateDate, dirtyBit, frozen, name, reference, referenceType, valid, parentid, versionEditor_id, null::boolean as automated, null::varchar as cwlPath, null::varchar as dockerfilePath, null::varchar as imageId, null::timestamp as lastBuilt, null::int8 as size, null::varchar as wdlPath, dagJson, isLegacyVersion, lastModified, subClass, toolTableJson, workflowPath, 2 as clazz_ from workflowversion ) version0_ where version0_.parentid=?`

Reply:
That's lame.  Is that all of the SQL for this, or are they also fetching all the sourcefiles subsequently, since those aren't lazy loaded in this branch?

You'd probably need something like `select count(*) from (select v.id from Version v where...` and I'm not even sure you can do a nested query like that in JPQL.

Just commenting, I did approve.

Reply:
i have a feeling that would make it worse